Reduce use of `const char*` in the code base

cdumez opened the pull request:
#### cb33335735a96497d668430668716a1b8f694e29
<pre>
Reduce use of `const char*` in the code base
<a href="https://bugs.webkit.org/show_bug.cgi?id=273277">https://bugs.webkit.org/show_bug.cgi?id=273277</a>

Reviewed by Darin Adler.

Reduce use of `const char*` in the code base by leveraging ASCIILiteral.

* Source/JavaScriptCore/assembler/MacroAssemblerCodeRef.h:
(JSC::MacroAssemblerCodeRef::dump const):
* Source/JavaScriptCore/ftl/FTLCompile.cpp:
(JSC::FTL::compile):
* Source/WTF/wtf/CodePtr.cpp:
(WTF::CodePtrBase::dumpWithName):
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::dumpWithName const):
(WTF::CodePtr::dump const):
* Source/WTF/wtf/Dominators.h:
(WTF::Dominators::Dominators):
(WTF::Dominators::ValidationContext::reportError):
* Source/WTF/wtf/Gigacage.h:
(Gigacage::name):
* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::toString):
(WTF::MemoryPressureHandler::setMemoryUsagePolicyBasedOnFootprint):
* Source/WTF/wtf/SingleRootGraph.h:
(WTF::SingleRootGraphNode::rootName):
(WTF::SingleRootGraphSet::dump const):
* Source/WTF/wtf/StringHashDumpContext.h:
(WTF::StringHashDumpContext::dump const):
* Source/WTF/wtf/SuspendableWorkQueue.cpp:
(WTF::SuspendableWorkQueue::stateString):
(WTF::SuspendableWorkQueue::suspend):
(WTF::SuspendableWorkQueue::resume):
* Source/WTF/wtf/SuspendableWorkQueue.h:
* Source/WTF/wtf/URLParser.cpp:
(WTF::serializeURLEncodedForm):
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
(WebCore::IDBServer::quotaErrorMessageName):
(WebCore::IDBServer::UniqueIDBDatabase::performCurrentOpenOperationAfterSpaceCheck):
(WebCore::IDBServer::UniqueIDBDatabase::createObjectStore):
(WebCore::IDBServer::UniqueIDBDatabase::renameObjectStore):
(WebCore::IDBServer::UniqueIDBDatabase::createIndex):
(WebCore::IDBServer::UniqueIDBDatabase::renameIndex):
(WebCore::IDBServer::UniqueIDBDatabase::putOrAddAfterSpaceCheck):
* Source/WebCore/Modules/push-api/PushDatabase.h:
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::errorCodeToString):
* Source/WebCore/Modules/webauthn/cbor/CBORReader.h:
* Source/WebCore/Modules/webauthn/fido/FidoConstants.cpp:
(fido::publicKeyCredentialTypeToString):
* Source/WebCore/Modules/webauthn/fido/FidoConstants.h:
* Source/WebCore/Modules/webdatabase/Database.cpp:
(WebCore::formatErrorMessage):
(WebCore::Database::performOpenAndVerify):
(WebCore::Database::incrementalVacuumIfNeeded):
* Source/WebCore/Modules/webdatabase/DatabaseTask.cpp:
(WebCore::DatabaseTask::performTask):
(WebCore::DatabaseOpenTask::debugTaskName const):
(WebCore::DatabaseCloseTask::debugTaskName const):
(WebCore::DatabaseTransactionTask::debugTaskName const):
(WebCore::DatabaseTableNamesTask::debugTaskName const):
* Source/WebCore/Modules/webdatabase/DatabaseTask.h:
* Source/WebCore/Modules/webdatabase/SQLTransaction.cpp:
(WebCore::SQLTransaction::performPendingCallback):
(WebCore::SQLTransaction::requestTransitToState):
(WebCore::SQLTransaction::scheduleCallback):
(WebCore::SQLTransaction::computeNextStateAndCleanupIfNeeded):
(WebCore::SQLTransaction::debugStepName):
* Source/WebCore/Modules/webdatabase/SQLTransaction.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionBackend.cpp:
(WebCore::SQLTransactionBackend::computeNextStateAndCleanupIfNeeded):
(WebCore::SQLTransactionBackend::requestTransitToState):
* Source/WebCore/Modules/webdatabase/SQLTransactionStateMachine.cpp:
(WebCore::nameForSQLTransactionState):
* Source/WebCore/Modules/webdatabase/SQLTransactionStateMachine.h:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::joinStrings):
* Source/WebCore/Modules/websockets/WebSocketExtensionParser.cpp:
(WebCore::isSeparator):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::name):
(WTF::printInternal):
* Source/WebCore/contentextensions/ContentExtensionActions.cpp:
(WebCore::ContentExtensions::makeJSString):
(WebCore::ContentExtensions::RedirectAction::RegexSubstitutionAction::applyToURL const):
* Source/WebCore/contentextensions/ContentExtensionParser.cpp:
(WebCore::ContentExtensions::getDomainList):
* Source/WebCore/contentextensions/DFABytecodeInterpreter.cpp:
(WebCore::ContentExtensions::DFABytecodeInterpreter::interpret):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::extractECDSASignatureInteger):
(WebCore::gcryptSign):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
(WebCore::extractEDDSASignatureInteger):
(WebCore::signEd25519):
* Source/WebCore/crypto/gcrypt/CryptoKeyECGCrypt.cpp:
(WebCore::curveName):
(WebCore::CryptoKeyEC::platformGeneratePair):
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
(WebCore::getRSAKeyParameter):
(WebCore::CryptoKeyRSA::algorithm const):
(WebCore::CryptoKeyRSA::exportData const):
* Source/WebCore/css/CSSCounterStyleRule.cpp:
(WebCore::CSSCounterStyleRule::cssText const):
* Source/WebCore/css/CSSTimingFunctionValue.cpp:
(WebCore::CSSStepsTimingFunctionValue::customCSSText const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
(WebCore::ShorthandSerializer::serializeLonghands const):
(WebCore::ShorthandSerializer::serializeLonghandsOmittingTrailingInitialValue const):
(WebCore::ShorthandSerializer::serializeGridArea const):
(WebCore::ShorthandSerializer::serializeGridRowColumn const):
(WebCore::ShorthandSerializer::serializeGridTemplate const):
* Source/WebCore/css/StyleColor.cpp:
(WebCore::StyleColor::colorFromAbsoluteKeyword):
* Source/WebCore/css/typedom/numeric/CSSNumericBaseType.h:
(WebCore::debugString):
* Source/WebCore/dom/Node.cpp:
(WebCore::stringForRareDataUseType):
(WebCore::Node::dumpStatistics):
(WebCore::appendAttributeDesc):
(WebCore::Node::showNode const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/html/FTPDirectoryDocument.cpp:
(WebCore::processFileDateString):
* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::policyTypeName):
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::logConsoleWarning):
(WebCore::logConsoleWarningForUpgrade):
* Source/WebCore/page/PerformanceLogging.cpp:
(WebCore::toString):
(WebCore::PerformanceLogging::memoryUsageStatistics):
(WebCore::PerformanceLogging::didReachPointOfInterest):
(WebCore::PerformanceLogging::getPlatformMemoryUsageStatistics):
* Source/WebCore/page/PerformanceLogging.h:
* Source/WebCore/page/cocoa/PerformanceLoggingCocoa.mm:
(WebCore::PerformanceLogging::getPlatformMemoryUsageStatistics):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::reportInvalidPathCharacter const):
* Source/WebCore/platform/RuntimeApplicationChecks.cpp:
(WebCore::processTypeDescription):
* Source/WebCore/platform/RuntimeApplicationChecks.h:
* Source/WebCore/platform/graphics/GraphicsContextState.cpp:
(WebCore::stateChangeName):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::isEqual):
(WebCore::CDMSessionAVContentKeySession::update):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::overrideUnauthorizedFunctions):
* Source/WebCore/rendering/RenderCounter.cpp:
(showCounterRendererTree):
* Source/WebCore/rendering/RenderCounter.h:
(showCounterRendererTree):
* Source/WebKit/Platform/IPC/HandleMessage.h:
(IPC::textStreamForLogging):

Canonical link: <a href="https://commits.webkit.org/278045@main">https://commits.webkit.org/278045@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec7f61490f48624e53d03125b96b1dede85d514d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49312 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28594 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52343 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/52157 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45415 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51616 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34608 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26214 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40287 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51413 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26141 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/42478 "Build was cancelled. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; 3 api tests failed or timed out; re-run-api-tests (cancelled)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21408 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23592 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43659 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7679 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/42614 "Build was cancelled. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Running apply-patch; Checked out pull request; Compiled JSC (warnings); jscore-tests (cancelled)") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45509 "Build is in progress. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; 5 api tests failed or timed out; re-run-api-tests (cancelled)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44162 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54060 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/48801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24394 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20577 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47609 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25666 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42686 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46612 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests (failure); re-run-api-tests (cancelled)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26505 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/56296 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7080 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25389 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11563 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -2925,9 +2925,7 @@ std::optional<KeyValuePair<String, String>> URLParser::parseQueryNameAndValue(St
-    const char* data = utf8.data();
-    for (size_t i = 0; i < utf8.length(); ++i) {
-        const char byte = data[i];
+    for (const char byte : utf8.span()) {
```
`const` here doesn't add anything

darinadler commented on Source/WebCore/contentextensions/ContentExtensionActions.cpp:
```diff
@@ -413,14 +413,14 @@ auto RedirectAction::RegexSubstitutionAction::deserialize(std::span<const uint8_
 }
 
-static JSRetainPtr<JSStringRef> makeJSString(const char* utf8)
+static JSRetainPtr<JSStringRef> makeJSString(ASCIILiteral utf8)
```
The name `utf8` no longer makes sense here, since ASCII literals are ASCII.

darinadler commented on Source/WebCore/html/FTPDirectoryDocument.cpp:
```diff
@@ -239,7 +239,7 @@ static String processFileDateString(const FTPTime& fileTime)
         return "Yesterday" + timeOfDay;
 
-    static const char* months[] = { "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "???" };
+    static const std::array months = { "Jan"_s, "Feb"_s, "Mar"_s, "Apr"_s, "May"_s, "Jun"_s, "Jul"_s, "Aug"_s, "Sep"_s, "Oct"_s, "Nov"_s, "Dec"_s, "???"_s };
```
`constexpr`

darinadler commented on Source/WebCore/page/PerformanceLogging.cpp:
```diff
@@ -39,39 +39,39 @@
+HashMap<ASCIILiteral, size_t> PerformanceLogging::memoryUsageStatistics(ShouldIncludeExpensiveComputations includeExpensive)
 {
-    HashMap<const char*, size_t> stats;
+    HashMap<ASCIILiteral, size_t> stats;
```
I don't understand why this is a map rather than a vector. Is it desirable to log in a random order? Is there a need to check uniqueness?

cdumez replied:
Not super familiar with this code. I just looked into it and it doesn't appear we need the check for uniqueness. I looked at the 2 call sites: 1 doesn't care about ordering because it merely populates a dictionary with the stats. The other one though does logging and thus ends up printing in random order. I think we can switch to a vector here.

cdumez committed:
Reduce use of `const char*` in the code base
https://bugs.webkit.org/show_bug.cgi?id=273277

Reviewed by Darin Adler.

Reduce use of `const char*` in the code base by leveraging ASCIILiteral.

* Source/JavaScriptCore/assembler/MacroAssemblerCodeRef.h:
(JSC::MacroAssemblerCodeRef::dump const):
* Source/JavaScriptCore/ftl/FTLCompile.cpp:
(JSC::FTL::compile):
* Source/WTF/wtf/CodePtr.cpp:
(WTF::CodePtrBase::dumpWithName):
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::dumpWithName const):
(WTF::CodePtr::dump const):
* Source/WTF/wtf/Dominators.h:
(WTF::Dominators::Dominators):
(WTF::Dominators::ValidationContext::reportError):
* Source/WTF/wtf/Gigacage.h:
(Gigacage::name):
* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::toString):
(WTF::MemoryPressureHandler::setMemoryUsagePolicyBasedOnFootprint):
* Source/WTF/wtf/SingleRootGraph.h:
(WTF::SingleRootGraphNode::rootName):
(WTF::SingleRootGraphSet::dump const):
* Source/WTF/wtf/StringHashDumpContext.h:
(WTF::StringHashDumpContext::dump const):
* Source/WTF/wtf/SuspendableWorkQueue.cpp:
(WTF::SuspendableWorkQueue::stateString):
(WTF::SuspendableWorkQueue::suspend):
(WTF::SuspendableWorkQueue::resume):
* Source/WTF/wtf/SuspendableWorkQueue.h:
* Source/WTF/wtf/URLParser.cpp:
(WTF::serializeURLEncodedForm):
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
(WebCore::IDBServer::quotaErrorMessageName):
(WebCore::IDBServer::UniqueIDBDatabase::performCurrentOpenOperationAfterSpaceCheck):
(WebCore::IDBServer::UniqueIDBDatabase::createObjectStore):
(WebCore::IDBServer::UniqueIDBDatabase::renameObjectStore):
(WebCore::IDBServer::UniqueIDBDatabase::createIndex):
(WebCore::IDBServer::UniqueIDBDatabase::renameIndex):
(WebCore::IDBServer::UniqueIDBDatabase::putOrAddAfterSpaceCheck):
* Source/WebCore/Modules/push-api/PushDatabase.h:
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::errorCodeToString):
* Source/WebCore/Modules/webauthn/cbor/CBORReader.h:
* Source/WebCore/Modules/webauthn/fido/FidoConstants.cpp:
(fido::publicKeyCredentialTypeToString):
* Source/WebCore/Modules/webauthn/fido/FidoConstants.h:
* Source/WebCore/Modules/webdatabase/Database.cpp:
(WebCore::formatErrorMessage):
(WebCore::Database::performOpenAndVerify):
(WebCore::Database::incrementalVacuumIfNeeded):
* Source/WebCore/Modules/webdatabase/DatabaseTask.cpp:
(WebCore::DatabaseTask::performTask):
(WebCore::DatabaseOpenTask::debugTaskName const):
(WebCore::DatabaseCloseTask::debugTaskName const):
(WebCore::DatabaseTransactionTask::debugTaskName const):
(WebCore::DatabaseTableNamesTask::debugTaskName const):
* Source/WebCore/Modules/webdatabase/DatabaseTask.h:
* Source/WebCore/Modules/webdatabase/SQLTransaction.cpp:
(WebCore::SQLTransaction::performPendingCallback):
(WebCore::SQLTransaction::requestTransitToState):
(WebCore::SQLTransaction::scheduleCallback):
(WebCore::SQLTransaction::computeNextStateAndCleanupIfNeeded):
(WebCore::SQLTransaction::debugStepName):
* Source/WebCore/Modules/webdatabase/SQLTransaction.h:
* Source/WebCore/Modules/webdatabase/SQLTransactionBackend.cpp:
(WebCore::SQLTransactionBackend::computeNextStateAndCleanupIfNeeded):
(WebCore::SQLTransactionBackend::requestTransitToState):
* Source/WebCore/Modules/webdatabase/SQLTransactionStateMachine.cpp:
(WebCore::nameForSQLTransactionState):
* Source/WebCore/Modules/webdatabase/SQLTransactionStateMachine.h:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::joinStrings):
* Source/WebCore/Modules/websockets/WebSocketExtensionParser.cpp:
(WebCore::isSeparator):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::name):
(WTF::printInternal):
* Source/WebCore/contentextensions/ContentExtensionActions.cpp:
(WebCore::ContentExtensions::makeJSString):
(WebCore::ContentExtensions::RedirectAction::RegexSubstitutionAction::applyToURL const):
* Source/WebCore/contentextensions/ContentExtensionParser.cpp:
(WebCore::ContentExtensions::getDomainList):
* Source/WebCore/contentextensions/DFABytecodeInterpreter.cpp:
(WebCore::ContentExtensions::DFABytecodeInterpreter::interpret):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::extractECDSASignatureInteger):
(WebCore::gcryptSign):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
(WebCore::extractEDDSASignatureInteger):
(WebCore::signEd25519):
* Source/WebCore/crypto/gcrypt/CryptoKeyECGCrypt.cpp:
(WebCore::curveName):
(WebCore::CryptoKeyEC::platformGeneratePair):
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
(WebCore::getRSAKeyParameter):
(WebCore::CryptoKeyRSA::algorithm const):
(WebCore::CryptoKeyRSA::exportData const):
* Source/WebCore/css/CSSCounterStyleRule.cpp:
(WebCore::CSSCounterStyleRule::cssText const):
* Source/WebCore/css/CSSTimingFunctionValue.cpp:
(WebCore::CSSStepsTimingFunctionValue::customCSSText const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
(WebCore::ShorthandSerializer::serializeLonghands const):
(WebCore::ShorthandSerializer::serializeLonghandsOmittingTrailingInitialValue const):
(WebCore::ShorthandSerializer::serializeGridArea const):
(WebCore::ShorthandSerializer::serializeGridRowColumn const):
(WebCore::ShorthandSerializer::serializeGridTemplate const):
* Source/WebCore/css/StyleColor.cpp:
(WebCore::StyleColor::colorFromAbsoluteKeyword):
* Source/WebCore/css/typedom/numeric/CSSNumericBaseType.h:
(WebCore::debugString):
* Source/WebCore/dom/Node.cpp:
(WebCore::stringForRareDataUseType):
(WebCore::Node::dumpStatistics):
(WebCore::appendAttributeDesc):
(WebCore::Node::showNode const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/html/FTPDirectoryDocument.cpp:
(WebCore::processFileDateString):
* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::policyTypeName):
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::logConsoleWarning):
(WebCore::logConsoleWarningForUpgrade):
* Source/WebCore/page/PerformanceLogging.cpp:
(WebCore::toString):
(WebCore::PerformanceLogging::memoryUsageStatistics):
(WebCore::PerformanceLogging::didReachPointOfInterest):
(WebCore::PerformanceLogging::getPlatformMemoryUsageStatistics):
* Source/WebCore/page/PerformanceLogging.h:
* Source/WebCore/page/cocoa/PerformanceLoggingCocoa.mm:
(WebCore::PerformanceLogging::getPlatformMemoryUsageStatistics):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::reportInvalidPathCharacter const):
* Source/WebCore/platform/RuntimeApplicationChecks.cpp:
(WebCore::processTypeDescription):
* Source/WebCore/platform/RuntimeApplicationChecks.h:
* Source/WebCore/platform/graphics/GraphicsContextState.cpp:
(WebCore::stateChangeName):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::isEqual):
(WebCore::CDMSessionAVContentKeySession::update):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::overrideUnauthorizedFunctions):
* Source/WebCore/rendering/RenderCounter.cpp:
(showCounterRendererTree):
* Source/WebCore/rendering/RenderCounter.h:
(showCounterRendererTree):
* Source/WebKit/Platform/IPC/HandleMessage.h:
(IPC::textStreamForLogging):

Canonical link: https://commits.webkit.org/278045@main
